bot/extensions/adventofcode/commands.py

```diff
@@ -0,0 +1,216 @@
+import logging
+import re
+from datetime import datetime
+
+import aiohttp
+import discord
+import pytz
+from bs4 import BeautifulSoup
+from discord import app_commands
+from discord.ext import commands
+
+from bot import core
+from bot.config import settings
+
+log = logging.getLogger(__name__)
+
+YEAR = datetime.now(tz=pytz.timezone("EST")).year
+API_URL = f"https://adventofcode.com/{YEAR}/leaderboard/private/view/975452.json"
+INTERVAL = 120
+AOC_REQUEST_HEADER = {"user-agent": "TWT AoC Event Bot"}
```

Comment:
```suggestion
AOC_REQUEST_HEADER = {"user-agent": "Tech With Tim Discord Bot https://github.com/SylteA/Discord-Bot"}
```

If we can put URLs in the user agent let's do something like this